### PR TITLE
Update google play location service version to 16.0.0

### LINKED ISSUE #198: Older Google play services version used on Kujaku
Currently Kujaku uses Google play services maps version 8.4.0  `com.google.android.gms:play-services-maps:8.4.0` and Google services location `com.google.android.gms:play-services-location:8.4.0`.

Google play services vision `com.google.android.gms:play-services-vision:17.0.2` was added to OpenSRP core to support Google barcode/qr readers.

Google services maps, location and vision services all use  
- com.google.android.gms:play-services-base

Google services maps and vision services all use 
- com.google.android.gms:play-services-basement
- com.google.android.gms:play-services-tasks

So in including both OpenSRP core and Kujaku in an application; the version of the 3 above libraries loaded is 16.0.1. This breaks Geowidget as its using older APIs.

Snapshot of play services dependencies from Kujaku after Importing OpenSRP core

image

### PR DESCRIPTION
**Pull request by ekigamba**
- Remove deprecated FuseLocationApi and use the new FusedLocationProviderClient

Fixes #198

### CODE REVIEW

File: library/src/main/java/io/ona/kujaku/location/clients/AndroidLocationClient.java [hunk 1]
```diff
@@ -165,8 +120,31 @@ public boolean isProviderEnabled() {
 
     @Override
     public void close() {
-        disconnectGoogleApiClient();
-
+        stopLocationUpdates();
         super.close();
     }
+
+    class AndroidLocationCallback extends LocationCallback {
```
Comment (vincent-karuri) [inline library/src/main/java/io/ona/kujaku/location/clients/AndroidLocationClient.java]: Add explicit scope to this.

File: library/src/main/java/io/ona/kujaku/location/clients/AndroidLocationClient.java [hunk 2]
```diff
@@ -165,8 +120,31 @@ public boolean isProviderEnabled() {
...
+            Location latestLocation = null;
+            for (Location location : locationResult.getLocations()) {
+                if (latestLocation == null || location.getTime() > latestLocation.getTime()) {
+                    latestLocation = location;
+                }
+            }
+
+            if (latestLocation != null) {
+                lastLocation = latestLocation;
+                if (getLocationListener() != null) {
```
Comment (vincent-karuri) [inline library/src/main/java/io/ona/kujaku/location/clients/AndroidLocationClient.java]: Why not combine this if statement with the one enclosing it and perform this `lastLocation = latestLocation` after calling the listener?

Comment (ekigamba) [reply library/src/main/java/io/ona/kujaku/location/clients/AndroidLocationClient.java]: So if `getLocationListener()` returns null for some reason, we will not update `lastLocation`?

Comment (vincent-karuri) [reply library/src/main/java/io/ona/kujaku/location/clients/AndroidLocationClient.java]: does it need to be updated in this case?

Comment (ekigamba) [reply library/src/main/java/io/ona/kujaku/location/clients/AndroidLocationClient.java]: Yea

Comment (vincent-karuri) [reply library/src/main/java/io/ona/kujaku/location/clients/AndroidLocationClient.java]: I think a better comment to explain why would be helpful.

Comment (vincent-karuri) [reply library/src/main/java/io/ona/kujaku/location/clients/AndroidLocationClient.java]: I don't think that's necessary.

**Pull request merged**